vertical-slice-architecture/README.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Vertical-Slice-Architecture
+aka: Layer-By-Feature
+category: Architectural
+language: en
+tag:
+- Decoupling
+---
+
+## Intent
+
+package the application based on features. Each feature will have its own set of layers (
+Models, Services, Repository and Controllers ).
+
+## Explanation
+
+> With vertical slice architecture we can have high cohesion within package and low coupling
+> among the packages. In Conceptual term
+
+> Consider that you are going to make a backend service for a online e-commerce application.
+> initially you make it with usual grouping of controllers, models etc. but as the application
+> grows more it's requires implementation of new features. Let's say that you thought of having
+> orders, customers and products associated layers. But now you need to include another set of
+> features with Cart system and wishlists. Now it's really hard to integrate those features it
+> requires lot's of dependency modifications and mocking. So if you make the package by feature
+> it will be really feasible for future additions. General example.
```

Comment:
For the explanation, please use the standard headings as described here: https://github.com/iluwatar/java-design-patterns/wiki/02.-Pattern-template Check the other patterns for examples as well.

Reply:
Updated, Do I need to have a code sample for this?.